src/Model.js

```diff
@@ -310,14 +311,32 @@ export default class Model {
    * too
    */
   @action
-  async destroy ({ optimistic = true }: DestroyOptions = {}): Promise<*> {
+  async destroy (
+    arg1: any,
+    arg2: any
+  ): Promise<*> {
+    let attributes
+    let optimistic
+
+    if (arguments.length === 0) {
+      attributes = {}
+      optimistic = true
+    } else if (arguments.length === 1) {
+      console.warn('Deprecation warning: destroy call requires a data argument if you are using options')
```

Comment:
hmm.. not sure. `model.delete(attr)` is a valid signature since options are optional. After seeing this I'm more sold on making the change **not** backward compatible and having a more consistent API.

Reply:
Bumping semver? It's what I said... :+1: I'll work on this.